editor/src/core/property-controls/property-controls-local.ts

```diff
@@ -74,6 +61,130 @@ async function parseInsertOption(
   }, parsedParams)
 }
 
+export const ComponentDescriptorFile = '/utopia/components.utopia.js'
+
+export function maybeUpdateComponentDescriptor(
+  parseResult: Array<ParseOrPrintResult>,
+  workers: UtopiaTsWorkers,
+  dispatch: EditorDispatch,
+) {
+  // TODO: check file extension here
+  // e.g. `.utopia.registration.js`
+  const componentsFile = parseResult.find((parse) => parse.filename === ComponentDescriptorFile)
+  if (componentsFile == null) {
+    return
+  }
+  if (!isParseSuccess(componentsFile.parseResult)) {
+    return
+  }
+  const combinedTopLevelArbitraryBlock = componentsFile.parseResult.combinedTopLevelArbitraryBlock
+
+  if (combinedTopLevelArbitraryBlock == null) {
+    return
+  }
+
+  try {
+    // TODO: this isn't the right function to call, as `resolveParamsAndRunJsCode` doesn't actually give us access to the exports/default exports
```

Comment:
I think we'd need to take a similar approach as the canvas here, where we using the `editor.codeResultCache.curriedRequireFn` with the current project contents to create a `require` function that we can use to evaluate a file: https://github.com/concrete-utopia/utopia/blob/2372876aec02dbb1f524a5adcbfc1d9e2bff0790/editor/src/components/canvas/ui-jsx-canvas.tsx#L387-L390